core/src/main/java/org/elasticsearch/cluster/ClusterStateTaskExecutor.java

```diff
@@ -38,6 +38,13 @@ default boolean runOnlyOnMaster() {
     }
 
     /**
+     * Callback invoked after new cluster state is published. Note that
+     * this method is not invoked if the cluster state was not updated.
+     */
+    default void clusterStatePublished(ClusterState newClusterState) {
```

Comment:
I'm doubtful. I wonder if we should do this or follow the proven path we have in the listener interface where we have a processed callback (changed or not). I'm leaning towards the later as it gives more options (with the burden of an extra equality check).